src/main/java/com/rocketden/main/controller/v1/RoomController.java

```diff
@@ -67,4 +69,9 @@ public RoomController(RoomService service) {
                                                       @RequestBody UpdateSettingsRequest request) {
         return new ResponseEntity<>(service.updateRoomSettings(roomId, request), HttpStatus.OK);
     }
+
+    @PostMapping("/rooms/{roomId}/spectator/{isSpectator}")
```

Comment:
nit: maybe change this to `/rooms/{roomId}/spectator` and include the spectator variable in the request body (SetSpectatorRequest), just to be more consistent with [REST naming conventions](https://restfulapi.net/resource-naming/)